jupyterhub/objects.py

```diff
@@ -3,7 +3,7 @@
 # Distributed under the terms of the Modified BSD License.
 import socket
 import warnings
-from urllib.parse import urlparse, urlunparse
+from urllib.parse import urljoin, urlparse, urlunparse
```

Comment:
Let's use `jupyterhub.utils.url_path_join` here, which I think is more likely to have the desired effect. `urljoin` behaves like relative URL resolution on a webpage, whereas `url_path_join` does a simpler path-style join.

For example:

```python
In [9]: urljoin("a/b", "c")
Out[9]: 'a/c'

In [10]: url_path_join("a/b", "c")
Out[10]: 'a/b/c'
```